Mathlib/Topology/Baire/BaireMeasurable.lean

```diff
@@ -0,0 +1,184 @@
+    BaireMeasurableSet (s ∪ t) :=
+  MeasurableSet.union hs ht
+
+theorem inter (hs : BaireMeasurableSet s) (ht : BaireMeasurableSet t) :
+    BaireMeasurableSet (s ∩ t) :=
+  MeasurableSet.inter hs ht
+
+theorem diff (hs : BaireMeasurableSet s) (ht : BaireMeasurableSet t) :
+    BaireMeasurableSet (s \ t) :=
+  MeasurableSet.diff hs ht
+
+theorem congr (hs : BaireMeasurableSet s) (h : s =ᵇ t) : BaireMeasurableSet t :=
+  EventuallyMeasurableSet.congr (α := α) hs h.symm
+
+end BaireMeasurableSet
+
+open Filter
+
+/--Any Borel set differs from some open set by a meager set. -/
+theorem MeasurableSet.residualEq_open [MeasurableSpace α] [BorelSpace α] (h : MeasurableSet s) :
+    ∃ u : Set α, (IsOpen u) ∧ s =ᵇ u := by
+  apply h.induction_on_open (fun s hs => ⟨s, hs, EventuallyEq.rfl⟩)
+  · rintro s - ⟨u, uo, su⟩
+    refine ⟨(closure u)ᶜ, isClosed_closure.isOpen_compl,
+      EventuallyEq.compl (su.trans $ EventuallyLE.antisymm subset_closure.eventuallyLE ?_)⟩
+    have : (u ∪ (closure u)ᶜ) ∈ residual _ :=
+      residual_of_dense_open (uo.union isClosed_closure.isOpen_compl) union_compl_closure_dense
+    rw[EventuallyLE]
+    convert this
+    simp only [le_Prop_eq, imp_iff_or_not]
+    rfl --terrible
+  rintro s - - hs
+  choose u uo su using hs
+  exact ⟨⋃ i, u i, isOpen_iUnion uo, EventuallyEq.countable_iUnion su⟩
+
+/--Any `BaireMeasurableSet` differs from some open set by a meager set. -/
+theorem BaireMeasurableSet.residualEq_open (h : BaireMeasurableSet s) :
+    ∃ u : Set α, (IsOpen u) ∧ s =ᵇ u := by
+  borelize α
+  rcases h with ⟨t, ht, hst⟩
+  rcases ht.residualEq_open with ⟨u, hu, htu⟩
+  exact ⟨u, hu, hst.trans htu⟩
+
+section Map
+
+open Set
+
+variable {f : α → β} (hc : Continuous f) (ho : IsOpenMap f)
+
+/-- The preimage of a `Dense` set under a continuous open map is `Dense`. -/
+theorem Dense.dense_preimage_of_isOpenMap {t : Set β} (ht : Dense t) : Dense (f ⁻¹' t) := by
+  rw [dense_iff_closure_eq, ← ho.preimage_closure_eq_closure_preimage hc,
+    dense_iff_closure_eq.mp ht, preimage_univ]
+
+theorem residual_map_le_of_isOpenMap : (residual α).map f ≤ residual β := by
+  apply le_countableGenerate_iff_of_countableInterFilter.mpr
+  rintro t ⟨ht, htd⟩
+  exact residual_of_dense_open (ht.preimage hc) (htd.dense_preimage_of_isOpenMap hc ho)
+
+/-- The preimage of a meager set under a continuous open map is meager. -/
+theorem IsMeagre.preimage_of_isOpenMap {s : Set β} (h : IsMeagre s) : IsMeagre (f ⁻¹' s) :=
+  residual_map_le_of_isOpenMap hc ho h
+
+/-- The preimage of a `BaireMeasurableSet` under a continuous open map is Baire measurable. -/
+theorem BaireMeasurableSet.preimage {s : Set β} (h : BaireMeasurableSet s) :
+    BaireMeasurableSet (f⁻¹' s) := by
+  rcases h with ⟨u, hu, hsu⟩
+  refine ⟨f ⁻¹' u, ?_, hsu.filter_mono <| residual_map_le_of_isOpenMap hc ho⟩
+  borelize α β
+  exact hc.measurable hu
+
+theorem Homeomorph.residual_map_eq (h : α ≃ₜ β) : (residual α).map h = residual β := by
```

Comment:
This belongs in the file where we prove other basic properties of `residual` (if `Homeomorph` is available there).

Reply:
Currently this is Topology/GDelta, which is strange. What do you think of adding in this PR another new file, say Topology/Baire/Basic, where the definitions and basic properties of `isMeager` go

Reply:
(Oh and `Homeomorph is in fact not available there)

Reply:
Does `Baire/Lemmas` have the required imports?

Reply:
No. The only import of that file is `Topology/GDelta`